Add initial methods and tests

sjwhalen opened the pull request.

sjwhalen committed:
Add requirements and setup files

sjwhalen committed:
Add module dataframe_show_reader and its tests

sjwhalen committed:
Add .idea/ to .gitignore

josh803316 reviewed (approved):
Looks good to me. Who else would you like to have check it out?

sjwhalen commented:
> Looks good to me. Who else would you like to have check it out?

As we subsequently discussed, this code has been looked at by several folks before, so I feel safe proceeding as is, but I also welcome feedback from any interested parties. I have posted a notice encouraging anyone who is interested to chime in and contribute. I'll wait to merge until Friday, August 9, 5 PM Pacific to give folks a chance to weigh in.

jdavenport committed:
Add support for arrays and arrays of arrays.

- Changed the data type line start/end indicators from '[]' to '{}'
  so we can use the '[]' with the new array data types.

- These data types are supported for both arrays and
  arrays of arrays: str, int, float, double

jdavenport committed:
Revert back to using '[]' as the data type line start/ends.

- Modified show_output_to_df() parser so it only consumes the first
  and last '[' and ']' - and thus, doesn't consume these characters in
  the string arrays.

jdavenport committed:
Change FloatType to DoubleType

jdavenport committed:
Support 'WrappedArray()' df.show() syntax.

- DataFrame.show() outputs arrays of arrays as:
  '[WrappedArray(1.0, 2.0), WrappedArray(1.0), WrappedArray()]'
  We convert that to [[1.0, 2.0], [1.0], []] and process
  as normal.

- Also, trim() array elements of spaces so we can handle arrays like:
  '[ hi ,bye, so long ]' which becomes: '[hi, bye, so long]'

jdavenport committed:
Change data types: array<string>, array<array<string>>

sjwhalen committed:
Merge branch 'feat/array-data-type-support' into add-initial-methods-and-tests

sjwhalen committed:
Reformat some code for consistency
